File: nagasena/src/org/openexi/proc/events/EXIEventSchemaNil.java
package org.openexi.proc.events;

import org.openexi.proc.common.BinaryDataSource;
import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventDescription;
import org.openexi.proc.common.XmlUriConst;
import org.openexi.schema.Characters;
import org.openexi.schema.EXISchemaConst;

public final class EXIEventSchemaNil implements EventDescription {

  private final EventType m_eventType;
  private final boolean m_nilled;
  private final String m_prefix;
  private final Characters m_lexicalValue;

  public EXIEventSchemaNil(boolean nilled, Characters lexicalValue, String prefix, EventType eventType) {
    assert eventType.itemType == EventType.ITEM_SCHEMA_NIL;
    m_nilled = nilled;
    m_lexicalValue = lexicalValue;
    m_eventType = eventType;
    m_prefix = prefix;
  }

  public boolean isNilled() {
    return m_nilled;
  }

  ///////////////////////////////////////////////////////////////////////////
  // Implementation of EXIEvent interface
  ///////////////////////////////////////////////////////////////////////////

  public byte getEventKind() {
    return EventDescription.EVENT_NL;
  }
  
  public String getURI() {
    return XmlUriConst.W3C_2001_XMLSCHEMA_INSTANCE_URI;
  }
  
  public String getName() {
    return "nil";
  }

  public int getNameId() {
    return EXISchemaConst.XSI_LOCALNAME_NIL_ID;
  }

  public int getURIId() {
    return XmlUriConst.W3C_2001_XMLSCHEMA_INSTANCE_URI_ID;
  }

  public String getPrefix() {
    return m_prefix;
  }

  /**
   * Returns the lexical value when lexical preservation is on, 
   * otherwise returns null. 
   */
  public Characters getCharacters() {
    return m_lexicalValue;
  }
  
  public BinaryDataSource getBinaryDataSource() {
    return null;
  }

  public EventType getEventType() {
    return m_eventType;
  }

}

File: nagasenacs/src/Nagasena/Proc/Events/EXIEventSchemaNil.cs
using System.Diagnostics;

using BinaryDataSource = Nagasena.Proc.Common.BinaryDataSource;
using EventType = Nagasena.Proc.Common.EventType;
using EventDescription = Nagasena.Proc.Common.EventDescription;
using EventDescription_Fields = Nagasena.Proc.Common.EventDescription_Fields;
using XmlUriConst = Nagasena.Proc.Common.XmlUriConst;
using Characters = Nagasena.Schema.Characters;
using EXISchemaConst = Nagasena.Schema.EXISchemaConst;

namespace Nagasena.Proc.Events {

  /// <exclude/>
  public sealed class EXIEventSchemaNil : EventDescription {

    private readonly EventType m_eventType;
    private readonly bool m_nilled;
    private readonly string m_prefix;
    private readonly Characters m_lexicalValue;

    public EXIEventSchemaNil(bool nilled, Characters lexicalValue, string prefix, EventType eventType) {
      Debug.Assert(eventType.itemType == EventType.ITEM_SCHEMA_NIL);
      m_nilled = nilled;
      m_lexicalValue = lexicalValue;
      m_eventType = eventType;
      m_prefix = prefix;
    }

    public bool Nilled {
      get {
        return m_nilled;
      }
    }

    ///////////////////////////////////////////////////////////////////////////
    // Implementation of EXIEvent interface
    ///////////////////////////////////////////////////////////////////////////

    public sbyte EventKind {
      get {
        return EventDescription_Fields.EVENT_NL;
      }
    }

    public string URI {
      get {
        return XmlUriConst.W3C_2001_XMLSCHEMA_INSTANCE_URI;
      }
    }

    public string Name {
      get {
        return "nil";
      }
    }

    public int NameId {
      get {
        return EXISchemaConst.XSI_LOCALNAME_NIL_ID;
      }
    }

    public int URIId {
      get {
        return XmlUriConst.W3C_2001_XMLSCHEMA_INSTANCE_URI_ID;
      }
    }

    public string Prefix {
      get {
        return m_prefix;
      }
    }

    /// <summary>
    /// Returns the lexical value when lexical preservation is on, 
    /// otherwise returns null. 
    /// </summary>
    public Characters Characters {
      get {
        return m_lexicalValue;
      }
    }

    public BinaryDataSource BinaryDataSource {
      get {
        return null;
      }
    }

    public EventType getEventType() {
      return m_eventType;
    }

  }

}